pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -282,6 +309,7 @@ func (r *Reconciler) updateCNISubnetConfig(ctx context.Context, cr *hubv1alpha1.
 		}
 		if !reflect.DeepEqual(cr.Status.CniSubnet, cniSubnet) {
 			cr.Status.CniSubnet = cniSubnet
+			cr.Status.RegistrationStatus = hubv1alpha1.RegistrationStatusRegistered
```

Comment:
Are we still checking for CniSubnet to set RegistrationStatusRegistered
Can't we check for Cluster Health check to set RegistrationStatusRegistered

Reply:
Currently the approach to validate the cluster is registered or not is through CNI subnet available. UI also requires CNI subnet to set status as registered. The slice creation also requires this field to be set. health check current only perform component health checks and doesn't provide overall healthcheck.

Reply:
@kmjayadeep @pnavali @bharath-avesha Can you please confirm on this?

Reply:
We will move this move of code once health check is implemented for cni subnet and node ips (going as tech debt) @rajendra-avesha